.github/workflows/ci.yml

```diff
@@ -437,4 +437,9 @@ jobs:
       - name: Prepare environment and run checks
         run:  |
           set -e
+          git remote add networkit "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY"
+          git fetch networkit
+          export CUR_BRANCH=$(git branch --show-current)
+          git checkout --track networkit/master
```

Comment:
Doesn't this break/fail if the current branch is named `master`, too (`fatal: A branch named 'master' already exists.`)? E.g., if I push to the master branch of my fork and then open a pull request from there or if the CI runs on the master branch? Wouldn't it be much simpler to use `networkit/master` in the diff command?

Reply:
Another thought: locally, this will fail of course unless you have a remote named `networkit`. Maybe you could first check if `networkit/master` exists and if not compare to `master`?